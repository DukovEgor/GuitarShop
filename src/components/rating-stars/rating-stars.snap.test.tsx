import { render } from '@testing-library/react';
import RatingStars from './rating-stars';

describe('Component: RatingStars', () => {
  test('should render correctly without additional props', () => {
    const { container } = render(<RatingStars RatingInt={0} />);
    expect(container).toMatchSnapshot();
  });

  test('should render correctly 3 red stars of 5 because rating is 3', () => {
    const ratingInt = 3;

    const { container } = render(<RatingStars RatingInt={ratingInt} />);
    expect(container).toMatchSnapshot();
  });

  test('should render correctly width of stars with different className', () => {
    const ratingInt = 3;

    const { container } = render(<RatingStars RatingInt={ratingInt} className='review__rating-panel' />);
    expect(container).toMatchSnapshot();
  });
});
