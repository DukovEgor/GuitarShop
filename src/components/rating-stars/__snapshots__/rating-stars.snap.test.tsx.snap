// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: RatingStars should render correctly 3 red stars of 5 because rating is 3 1`] = `
<div>
  <svg
    aria-hidden="true"
    height="11"
    width="12"
  >
    <use
      xlink:href="#icon-full-star"
    />
  </svg>
  <svg
    aria-hidden="true"
    height="11"
    width="12"
  >
    <use
      xlink:href="#icon-full-star"
    />
  </svg>
  <svg
    aria-hidden="true"
    height="11"
    width="12"
  >
    <use
      xlink:href="#icon-full-star"
    />
  </svg>
  <svg
    aria-hidden="true"
    height="11"
    width="12"
  >
    <use
      xlink:href="#icon-star"
    />
  </svg>
  <svg
    aria-hidden="true"
    height="11"
    width="12"
  >
    <use
      xlink:href="#icon-star"
    />
  </svg>
</div>
`;

exports[`Component: RatingStars should render correctly width of stars with different className 1`] = `
<div>
  <svg
    aria-hidden="true"
    height="16"
    width="16"
  >
    <use
      xlink:href="#icon-full-star"
    />
  </svg>
  <svg
    aria-hidden="true"
    height="16"
    width="16"
  >
    <use
      xlink:href="#icon-full-star"
    />
  </svg>
  <svg
    aria-hidden="true"
    height="16"
    width="16"
  >
    <use
      xlink:href="#icon-full-star"
    />
  </svg>
  <svg
    aria-hidden="true"
    height="16"
    width="16"
  >
    <use
      xlink:href="#icon-star"
    />
  </svg>
  <svg
    aria-hidden="true"
    height="16"
    width="16"
  >
    <use
      xlink:href="#icon-star"
    />
  </svg>
</div>
`;

exports[`Component: RatingStars should render correctly without additional props 1`] = `
<div>
  <svg
    aria-hidden="true"
    height="11"
    width="12"
  >
    <use
      xlink:href="#icon-star"
    />
  </svg>
  <svg
    aria-hidden="true"
    height="11"
    width="12"
  >
    <use
      xlink:href="#icon-star"
    />
  </svg>
  <svg
    aria-hidden="true"
    height="11"
    width="12"
  >
    <use
      xlink:href="#icon-star"
    />
  </svg>
  <svg
    aria-hidden="true"
    height="11"
    width="12"
  >
    <use
      xlink:href="#icon-star"
    />
  </svg>
  <svg
    aria-hidden="true"
    height="11"
    width="12"
  >
    <use
      xlink:href="#icon-star"
    />
  </svg>
</div>
`;
